### PR TITLE
Add MetricAlarm resource type

### PR DESCRIPTION
**Pull request by jessebye**
Issue #, if available: n/a

Description of changes: Adds the `MetricAlarm` resource type, with functioning e2e test.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/metric_alarm/sdk.go [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+			ko.Spec.Metrics = f15
+		} else {
+			ko.Spec.Metrics = nil
+		}
+		if elem.Namespace != nil {
+			ko.Spec.Namespace = elem.Namespace
+		} else {
+			ko.Spec.Namespace = nil
+		}
+		if elem.OKActions != nil {
```
[the remark this answers is not in the document]
Comment (jessebye) [reply pkg/resource/metric_alarm/sdk.go:239]: I tried this, but it didn't make any difference for the `sdk.go` that got generated :thinking:

[the remark this answers is not in the document]
Comment (a-hilaly) [reply pkg/resource/metric_alarm/sdk.go:239]: Discussed offline with @jessebye about this. AlarmName should be an immutable field we can ignore this for now.

**Review by a-hilaly — APPROVED** [review]
Thank you @jessebye :+1:
/lgtm

**Pull request merged**